main.py

```diff
@@ -80,18 +80,23 @@ def get_role(server_roles, target_id):
     botspam_channel = client.get_channel(int(config.bot_spam))
     storytime_channel = client.get_channel(int(config.story_time))
 
-    # Check if the message author has the Game Master role
     isGameMaster = False
-    if message.guild == gamelog_channel.guild:
-        if game_master in [y.id for y in message.guild.get_member(message.author.id).roles]:
-            isGameMaster = True
-
     isAdmin = False
-    if message.guild == gamelog_channel.guild:
-        if administrator in [y.id for y in message.guild.get_member(message.author.id).roles]:
-            isAdmin = True
-
-    result = process(message,isGameMaster,isAdmin)
+    isPeasant = False
+    try:
+        if message.guild == gamelog_channel.guild:
+            role_table = [y.id for y in message.guild.get_member(message.author.id).roles]
+
+            if game_master in role_table:
+                isGameMaster = True
+            if administrator in role_table:
+                isAdmin = True
+            if peasant in role_table and message.author.bot == True:
```

Comment:
is `peasant` defined anywhere? Also, why are you checking if a bot is in the role table?!

Reply:
It's defined in `config.py`. And we're not checking if there's a bot in the role table, we're checking that the user is a bot AND has the assigned bot role.
The bots can influence the bot quite heavily with those commands, so a double layer of protection wouldn't hurt.